src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,131 @@
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import duke.exception.DukeException;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.TaskList;
+import duke.tasks.ToDo;
+
+/**
+ * A class that represents the file storage of list of task.
+ */
+public class Storage {
+    private String filePath;
+
+    /**
+     * Constructs the file Storage.
+     *
+     * @param filePath the file path to store the Task List.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Returns a list of tasks that is read from the file storage.
+     *
+     * @return a list of tasks that is read from the file storage.
+     * @throws DukeException if the file is not read correctly.
+     */
+    public List<Task> readFromFile() throws DukeException {
+        List<Task> tasks = new ArrayList<>();
+
+        try {
+            File file = new File(filePath);
+            file.createNewFile();
+            Scanner sc = new Scanner(file);
+
+            while (sc.hasNext()) {
+                Task taskToRead;
+                String line = sc.nextLine();
+                String[] instructions = line.split(" \\| ");
+                int num = instructions.length;
+                String type = instructions[0].strip();
+                if (type.equals("T")) {
+                    if (num != 3) {
+                        throw new DukeException("The data format of the file is incorrect\n");
+                    }
+                    taskToRead = new ToDo(instructions[2].strip());
+                    if (instructions[1].strip().equals("1")) {
+                        taskToRead.markAsDone();
+                    }
+                    tasks.add(taskToRead);
+                } else if (type.equals("D")) {
+                    if (num != 4) {
+                        throw new DukeException("The data format of the file is incorrect\n");
+                    }
+                    LocalDateTime deadlineDateTime = LocalDateTime.parse(instructions[3].strip());
+                    taskToRead = new Deadline(instructions[2].strip(), deadlineDateTime);
+                    if (instructions[1].strip().equals("1")) {
+                        taskToRead.markAsDone();
+                    }
+                    tasks.add(taskToRead);
+                } else if (type.equals("E")) {
+                    if (num != 4) {
+                        throw new DukeException("The data format of the file is incorrect\n");
+                    }
+                    LocalDateTime eventDateTime = LocalDateTime.parse(instructions[3].strip());
+                    taskToRead = new Event(instructions[2].strip(), eventDateTime);
+                    if (instructions[1].strip().equals("1")) {
+                        taskToRead.markAsDone();
+                    }
+                    tasks.add(taskToRead);
```

Comment:
how would it handle the case when type does not equal T, D, or E? perhaps an else block would help!

Reply:
Thank you Qifeng for your helpful comment and review. I will make changes according to your review suggestions.